### PR TITLE
Enabled traversing multiple buffers at once

### PR DESCRIPTION
**Pull request by matt-bourke**
This aims to address https://github.com/helix-editor/helix/issues/10400 and allows prefixing `gn` and `gp` commands with a number to cycle through multiple buffers.

Feedback is always welcome.

This is my first attempt at contributing to something, so go easy on me 😅

### CODE REVIEW

**Review by archseer — APPROVED** [review]
Much cleaner!

File: helix-term/src/commands.rs [hunk 1]
```diff
@@ -799,28 +799,32 @@ fn goto_line_start(cx: &mut Context) {
...
-            let mut iter = iter.rev().skip_while(|id| *id != &current);
-            iter.next(); // skip current item
-            iter.next().or_else(|| editor.documents.keys().next_back())
+            // skip 'count' times past current buffer
+            let mut iter = iter
+                .rev()
+                .cycle()
+                .skip_while(|id| *id != &current)
+                .skip(count);
+            iter.next()
```
Comment (pascalkuthe) [inline helix-term/src/commands.rs:827]:
```suggestion
            iter
                .rev()
                .cycle()
                .skip_while(|id| *id != &current)
                .nth(count);
```

File: helix-term/src/commands.rs [hunk 2]
```diff
@@ -799,28 +799,32 @@ fn goto_line_start(cx: &mut Context) {
...
 
     let id = match direction {
         Direction::Forward => {
             let iter = editor.documents.keys();
-            let mut iter = iter.skip_while(|id| *id != &current);
-            iter.next(); // skip current item
-            iter.next().or_else(|| editor.documents.keys().next())
+            // skip 'count' times past current buffer
+            let mut iter = iter.cycle().skip_while(|id| *id != &current).skip(count);
+            iter.next()
```
Comment (pascalkuthe) [inline helix-term/src/commands.rs:817]:
```suggestion
            iter.cycle().skip_while(|id| *id != &current).nth(count);
```

**Review by pascalkuthe — COMMENTED** [review]
yeah much nicer, one nit to make it even more concise is to use `nth` instead of `skip(n).next()`

**Review by the-mikedavis — APPROVED** [review]

**Review by pascalkuthe — APPROVED** [review]

**Pull request merged**